tensorflow/core/common_runtime/executor.cc

```diff
@@ -2215,20 +2215,7 @@ bool ExecutorState::NodeDone(const Status& s, const Node* node,
   nodestats::SetAllEnd(stats);
   if (stats) {
     if (stats_collector_) {
-#if TENSORFLOW_USE_DIRECTML
```

Comment:
@adtsai Is there a reason this change was needed in the first place? I couldn't find where this is being parsed or being used for DML-specific stuff. If it's being used somewhere, it would likely have stopped working anyway when you changed physical_device_desc to be json data instead of a simple string.

Reply:
This was done for the benefit of chrome traces. After the json change the string looks a little different, but it still shows up in the trace:

![image](https://user-images.githubusercontent.com/22506766/104654646-4a227f00-5671-11eb-8957-158acde47f58.png)

What gets broken by this change?

Reply:
I don't think we're meant to modify this string at all since it's not only visual but is used logically in the code. This "name" is used as the key in a dictionary in the cost model graph logic failing comparisons when compared against the device strings in other dictionaries (e.g. the vanilla /device/DML:0 strings). See StepStatsCollector::FinalizeInternal in step_stats_collector.cc. FunctionTest.testSwapImplementationWithGrapplerPlugin was failing because of this.

Reply:
Hmm, interesting. I didn't know this was used for any sort of logic. I guess we'll have to go without prettified device names in the chrome traces.